prime-angular-frontend/src/app/core/services/abstract-logger.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { environment } from '@env/environment.prod.template';
+
+@Injectable({
+  providedIn: 'root'
+})
+export abstract class AbstractLoggerService {
+  constructor() { }
+
+  /**
+   * @description
+   * General output of logging information.
+   */
+  public log(msg: string, ...data: any[]) {
+    this.send('log', { msg, data });
+  }
+
+  /**
+   * @description
+   * Informative output of logging information.
+   */
+  public info(msg: string, ...data: any[]) {
+    this.send('info', { msg, data });
+  }
+
+  /**
+   * @description
+   * Outputs a warning message.
+   */
+  public warn(msg: string, ...data: any[]) {
+    this.send('warn', { msg, data });
+  }
+
+  /**
+   * @description
+   * Outputs an error message.
+   */
+  public error(msg: string, ...data: any[]) {
+    this.send('error', { msg, data });
+  }
+
+  /**
+   * @description
+   * Outputs a stack trace.
+   */
+  public trace(msg: string, ...data: any[]) {
+    this.send('error', { msg, data });
+  }
```

Comment:
`trace` and `log` are not referenced anywhere ... do we still need these?

Reply:
Just cloned the code from the previous logger. If they aren't called I can remove